classes/migration/upgrade/v3_4_0/I8073_RemoveNotesWithoutQueriesAndRelatedObjects.inc.php

```diff
@@ -0,0 +1,134 @@
+            $notesFileRows = DB::table('submission_files as sf')
+                ->join('files as f', 'sf.file_id', '=', 'f.file_id')
+                ->where('sf.assoc_type', '=', self::ASSOC_TYPE_NOTE)
+                ->where('sf.assoc_id', '=', $noteId)
+                ->where('sf.file_stage', '=', self::SUBMISSION_FILE_QUERY)
+                ->get([
+                        'sf.submission_file_id as submissionFileId',
+                        'sf.file_id as fileId',
+                        'f.path as filePath'
+                    ]);
+
+            foreach ($notesFileRows as $submissionFileRow) {
+                $submissionFileId = $submissionFileRow->submissionFileId;
+                $submissionFileFileId = $submissionFileRow->fileId;
+                $submissionFilePath = $submissionFileRow->filePath;
+
+                error_log("Removing submission file revisions that relate to the submission file entry ID ${submissionFileId} which will be deleted.");
+                DB::table('submission_file_revisions')
+                    ->where('submission_file_id', '=', $submissionFileId)
+                    ->delete();
+
+                error_log("Removing review round files that relate to the submission file entry ID ${submissionFileId} which will be deleted.");
+                DB::table('review_round_files')
+                    ->where('submission_file_id', '=', $submissionFileId)
+                    ->delete();
+
+                error_log("Removing review files that relate to the submission file entry ID ${submissionFileId} which will be deleted.");
+                DB::table('review_files')
+                    ->where('submission_file_id', '=', $submissionFileId)
+                    ->delete();
+
+                // Create a Filesystem object with the appropriate adapter to access the actual files
+                $umask = Config::getVar('files', 'umask', 0022);
+                $adapter = new LocalFilesystemAdapter(
+                    Config::getVar('files', 'files_dir'),
+                    PortableVisibilityConverter::fromArray([
+                        'file' => [
+                            'public' => self::FILE_MODE_MASK & ~$umask,
+                            'private' => self::FILE_MODE_MASK & ~$umask,
+                        ],
+                        'dir' => [
+                            'public' => self::DIRECTORY_MODE_MASK & ~$umask,
+                            'private' => self::DIRECTORY_MODE_MASK & ~$umask,
+                        ]
+                    ]),
+                    LOCK_EX,
+                    LocalFilesystemAdapter::DISALLOW_LINKS
+                );
+
+                $filesystem = new Filesystem($adapter);
+
+                if ($filesystem->has($submissionFilePath)) {
+                    error_log("A submission file was found at ${submissionFilePath}. Trying to delete it...");
```

Comment:
There's no need for an `error_log` in this case; there are logs in both conditions below.

Reply:
Removed